sets/maps.go

```diff
@@ -44,8 +44,14 @@ func IntersectMapKeys[K comparable, V any](maps ...map[K]V) map[K]V {
 	}
 	smallestMap := maps[smallestIdx]
 	final := make(map[K]V, len(smallestMap))
-	for k, v := range smallestMap {
-		final[k] = v
+	for k := range smallestMap {
```

Comment:
My impression is that this is perhaps over-elaborate compared to https://github.com/abcxyz/pkg/pull/339/files (5 for loops vs. 3). Is there a reason to prefer this approach?

Reply:
Yeah, this is somewhat difficult to parse IMO.